client/src/components/ImageUploadContainer.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { FormattedMessage } from 'react-intl';
+import { uploadImage, fetchImages } from '../store/actions/ImagesActions';
+
+interface State {
+  image: File | null;
+}
+
+interface Props {
+  uploadImage(image: File): Promise<void>;
+  fetchImages(): Promise<void>;
+}
+
+class ImageUploadContainerView extends React.PureComponent<Props, State> {
+  state: State = { image: null };
+  triggerUpload = (event: React.MouseEvent) => {
+    if (this.state.image) {
+      this.props
```

Comment:
I'd go with async/await or `this.props.uploadImage(this.state.image).then(this.props.fetchImages)`